crates/client/rpc/src/lib.rs

```diff
@@ -196,13 +199,81 @@ where
             Err(e) => Err(e.into()),
         }
     }
+
+    fn get_block_with_tx_hashes(&self, block_id: StarknetBlockId) -> RpcResult<MaybePendingBlockWithTxHashes> {
+        let substrate_block_hash = match block_id {
+            StarknetBlockId::BlockHash(h) => madara_backend_client::load_hash(
+                self.client.as_ref(),
+                &self.backend,
+                H256::from_str(&h).map_err(|e| {
+                    error!("Failed to convert '{h}' to H256: {e}");
+                    StarknetRpcApiError::BlockNotFound
+                })?,
+            )
+            .map_err(|e| {
+                error!("Failed to load Starknet block hash for Substrate block with hash '{h}': {e}");
+                StarknetRpcApiError::BlockNotFound
+            })?,
+            StarknetBlockId::BlockNumber(n) => {
+                self.client.hash(UniqueSaturatedInto::unique_saturated_into(n)).map_err(|e| {
+                    error!("Failed to retrieve the hash of block number '{n}': {e}");
+                    StarknetRpcApiError::BlockNotFound
+                })?
+            }
+            StarknetBlockId::BlockTag(t) => match t {
+                mc_rpc_core::BlockTag::Latest => Some(self.client.info().best_hash),
+                mc_rpc_core::BlockTag::Pending => None,
+            },
+        }
+        .ok_or(StarknetRpcApiError::BlockNotFound)?;
+
+        let block = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .current_block(substrate_block_hash)
+            .unwrap_or_default();
+
+        let txs_hashes = block.transactions_hashes().iter().map(|tx| tx.to_string()).collect();
+        let sequencer_address =
+            String::from_utf8(hex::encode(block.header().sequencer_address).into()).map_err(|e| {
+                error!("SequencerAddress: Failed to convert 'sequencer_address' attribute to String: {e}");
+                StarknetRpcApiError::InternalServerError
+            })?;
+
+        let block_with_tx_hashes = BlockWithTxHashes {
+            transactions: txs_hashes,
+            // TODO: Status hardcoded, get status from block
+            status: BlockStatus::Pending,
```

Comment:
Where could I get it from? I think it is not implemented, right?

Reply:
I don't think so, basically if it is in `current_block` mapping then it means it's been "accepted on L2". Pending transactions are stored in the "pending" storage map. Probably we could add an internal function in the pallet that constructs the pending block out of this mapping or you can do it in a helper function client-side.

Reply:
That makes sense. If you think it's necessary, I can add that method to the pallet and the runtime. It would return all pending transactions that have not yet been assigned a block, and return the `PendingBlockWithTxHashes` struct:

```
pub struct PendingBlockWithTxHashes {
    /// The hashes of the transactions included in this block
    pub transactions: Vec<FieldElement>,
    /// The time in which the block was created, encoded in Unix time
    pub timestamp: u64,
    /// The Starknet identity of the sequencer submitting this block
    pub sequencer_address: FieldElement,
    /// The hash of this block's parent
    pub parent_hash: FieldElement,
}
```